src/app/components/repos-search/repos-search.component.ts

```diff
@@ -36,18 +40,31 @@ export class ReposSearchComponent {
   ) {}
 
   hasQuery(): boolean {
-    return !!this.queryInputValue;
+    return this.queryInputValue.length > 0;
   }
 
   ngOnInit() {
     this.searchForm = this.formBuilder.group({
       query: [''],
     });
+
+    this.termService.search(this.queryValue);
+    this.queryInputValue = this.queryValue;
+  }
+
+  ngAfterViewInit() {
+    if (this.autofocus) {
+      this.queryElement.nativeElement.focus();
+    }
   }
 
   onKey(event: any) {
     this.queryInputValue = event.target.value;
     this.termService.search(this.queryInputValue);
+
+    if (event.code === 'Escape' || event.code === 'Enter') {
```

Comment:
Does this handle when the textbox loses focus? Do we want it to keep showing the list when it loses focus?

Reply:
On the `<input>` element, I added a blur listener to call `hideAutocomplete()`, so we should be handled in that case. Good thinking, though!